services/provider/server/consumer.go

```diff
@@ -224,3 +226,40 @@ func (c *ocsConsumerManager) Get(ctx context.Context, id string) (*ocsv1alpha1.S
 
 	return consumerObj, nil
 }
+
+func (c *ocsConsumerManager) UpdateStatusLastHeatbeat(ctx context.Context, id string) error {
+	uid := types.UID(id)
+
+	c.mutex.RLock()
+	consumerName, ok := c.nameByUID[uid]
+	if !ok {
+		c.mutex.RUnlock()
+		klog.Warningf("no storageConsumer found with UID %q", id)
+		return nil
+	}
+	c.mutex.RUnlock()
+
+	patchInfo := struct {
+		Op    string      `json:"op"`
+		Path  string      `json:"path"`
+		Value interface{} `json:"value"`
+	}{
+		Op:    "replace",
+		Path:  "/status/lastHeartbeat",
+		Value: metav1.Now(),
+	}
+	jsonPatchInfo, _ := json.Marshal([]interface{}{patchInfo})
+	patch := client.RawPatch(types.JSONPatchType, jsonPatchInfo)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I don't think we need RetryOnCOnflict here as there cannot be conflicts when we use patch as the entire operation is atomic to the k8s API.

Reply:
Updated in the latest commit